### PR TITLE
[release-1.2] ⚠️ Use Kubernetes 1.26 in Quick Start docs and CAPD

### PR DESCRIPTION
**Pull request by aniruddha2000**
Signed-off-by: Aniruddha Basak <codewithaniruddha@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
part of #7671 

cc @kubernetes-sigs/cluster-api-release-team

### CODE REVIEW

File: test/e2e/clusterctl_upgrade_test.go [hunk 1]
```diff
@@ -26,11 +26,96 @@ import (
 var _ = Describe("When testing clusterctl upgrades [clusterctl-Upgrade]", func() {
 	ClusterctlUpgradeSpec(ctx, func() ClusterctlUpgradeSpecInput {
 		return ClusterctlUpgradeSpecInput{
-			E2EConfig:             e2eConfig,
```
Comment (killianmuldoon) [inline test/e2e/clusterctl_upgrade_test.go:29]: Looks like there's a diff between the way this is in 1.2 vs 1.3 - I think the solution is to stick with whatever version is current in v1.2 - it looks like there's no Kubernetes version changes needed here?

Comment (aniruddha2000) [reply test/e2e/clusterctl_upgrade_test.go]: Yep!

File: .github/ISSUE_TEMPLATE/kubernetes_bump.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+---
```
Comment (sbueringer) [inline .github/ISSUE_TEMPLATE/kubernetes_bump.md:1]: Let's drop this file from this PR. We only actually need the template on main and it doesn't have to exist on release-1.2

File: Tiltfile [hunk 3]
```diff
@@ -1,6 +1,20 @@
 # -*- mode: Python -*-
 
 load("ext://uibutton", "cmd_button", "text_input")
+envsubst_cmd = "./hack/tools/bin/envsubst"
```
Comment (sbueringer) [inline Tiltfile:4]: Let's drop the diff from this PR. The release-1.2 Tiltfile worked quite a bit differently and it's not super important that we use 1.26 there

File: docs/book/src/developer/tilt.md [hunk 4]
```diff
@@ -259,6 +259,42 @@ configs](https://github.com/kubernetes-sigs/cluster-api/tree/main/test/infrastru
 These can be customized for your specific needs.
 
 <aside class="note">
+### Deploying a workload cluster
```
Comment (sbueringer) [inline docs/book/src/developer/tilt.md:262]: Same here - let's also drop this diff

**Comment by fabriziopandini** [comment]
/hold 
after tomorrow's patch release; this will slit to next month's patch release (we need CI signal for it first)

File: Tiltfile [hunk 5]
```diff
@@ -2,6 +2,20 @@
 
 load("ext://uibutton", "cmd_button", "text_input")
 
+clusterctl_cmd = "./bin/clusterctl"
```
Comment (sbueringer) [inline Tiltfile:5]: @aniruddha2000 Looks like this delta is still there :)

**Comment by sbueringer** [comment]
/lgtm

/hold

Will merge through after the patch releases

**Pull request merged**